### DISCUSSION TITLE
How to run NNI on TPU

### DISCUSSION THREAD

**Discussion by H-arshit** [microsoft/nni#3772] [Q&A] [answered]
I want to run my trials on TPU. What changes do i have to make in config.yml.

**Comment by QuanluZhang** [comment, marked as answer]
Hi @H-arshit , I have not used TPU, but I assume it is provided within a VM. Then it would be pretty easy to run NNI on TPU through NNI's `remote machine` mode. In your scenario, let's say you have 3 TPU-VMs (T1, T2, T3), and another VM T0. Note that T1/T2/T3 and T0 should be reachable to each other. Then you can create an nni experiment on T0, and the yaml config file of this experiment should include the configure (e.g., username, password) of the 3 TPU-VMs. [Here](https://nni.readthedocs.io/en/latest/TrainingService/RemoteMachineMode.html) is a detailed document of how to setup `remote machine` mode.

**Comment by ultmaster** [comment]
@H-arshit Any updates?

**Comment by scarlett2018** [comment]
> Hi @H-arshit , I have not used TPU, but I assume it is provided within a VM. Then it would be pretty easy to run NNI on TPU through NNI's `remote machine` mode. In your scenario, let's say you have 3 TPU-VMs (T1, T2, T3), and another VM T0. Note that T1/T2/T3 and T0 should be reachable to each other. Then you can create an nni experiment on T0, and the yaml config file of this experiment should include the configure (e.g., username, password) of the 3 TPU-VMs. [Here](https://nni.readthedocs.io/en/latest/TrainingService/RemoteMachineMode.html) is a detailed document of how to setup `remote machine` mode.

@H-arshit any feedback? thanks.
